src/proof/PGInterpolator.cc

```diff
@@ -538,14 +538,23 @@ void ProofGraph::produceSingleInterpolant ( vec<PTRef> &interpolants, const ipar
                 std::map<PTRef, icolor_t> ptref2label;
                 std::vector<Lit>& cl = n->getClause();
 
-                for(std::size_t j = 0; j < cl.size(); ++j)
-                {
+                for(std::size_t j = 0; j < cl.size(); ++j) {
                     ptref2label[varToPTRef(var(cl[j]))] = getVarColor(n, var(cl[j]));
                 }
 
                 partial_interp = thandler->getInterpolant (A_mask, &ptref2label, pmanager);
                 clearTSolver();
             }
+            else if (n->getType() == clause_type::CLA_SPLIT) {
+                auto const & clause = n->getClause();
+                assert(clause.size() == 2); // only binary splits at the moment
+                auto color = getVarColor(n, var(clause[0]));
+                assert(color == getVarColor(n, var(clause[1]))); // same theory variables in the atoms of the split => same color
+                assert(color == I_A || color == I_B || color == I_AB);
+                // If split on A-local (B-local) term, then return False (True). This is the same as in purely propoositional case.
+                // If split on AB-shared term, we can choose if we treat it as A-clause (resulting in False) or B-clause (resulting in True). We arbitrarily choose A now.
+                partial_interp = color == I_A ? logic_.getTerm_false() : (color == I_B ? logic_.getTerm_true() : logic_.getTerm_false());
+            }
```

Comment:
This was the missing part, right?

Reply:
Yes, but not only this. I added the clause type `clause_type::CLA_SPLIT`.